ksqldb-execution/src/main/java/io/confluent/ksql/execution/interpreter/TermEvaluationContext.java

```diff
@@ -16,10 +16,14 @@
 package io.confluent.ksql.execution.interpreter;
 
 import io.confluent.ksql.GenericRow;
+import java.util.Deque;
+import java.util.LinkedList;
+import java.util.Map;
 
 public final class TermEvaluationContext {
 
   private final GenericRow row;
+  private final Deque<Map<String, Object>> deque = new LinkedList<>();
```

Comment:
nit: lambdaArgumentsMappingDeque?

Reply:
I ended up calling it `variableMappingsStack` since it's being used to mimic lambda call stacks.